docs/nlp.md

```diff
@@ -17,22 +17,109 @@ and records those findings alongside the coded FHIR data.
 
 This way you can often surface symptoms that simply aren't recorded in the traditional FHIR data.
 
-## NLP Is Always Specific to a Study
+## NLP Is Always Specific to a Clinical Purpose
 
 The first thing to understand is that Cumulus ETL always
-runs NLP in the context of a specific study purpose.
+runs NLP in the context of a specific clinical purpose,
+which we'll call a "study."
 
 Each study's design will have its own needs and its own NLP strategy,
-so we support multiple approaches.
+so Cumulus ETL supports multiple approaches.
 
 {: .note }
 Example: The `covid_symptom` study uses cTAKES and a negation transformer working together to tag
 COVID-19 symptoms in clinical notes.
 But another study might use the Llama2 large language model,
 with a prompt like "Does this patient have a nosebleed?"
 
+### But With Re-Usable Code
+
+While the clinical "business logic" of how to drive NLP is inevitably study-specific,
+the code structure of Cumulus ETL is generic.
+That's what makes it easy to support multiple different NLP strategies.
+
+We'll go into more depth about what an NLP task does under the covers [below](#technical-workflow).
+But for now, here's a basic outline of how Cumulus ETL runs an NLP study task:
+1. Prepare the clinical notes
+2. Hand those notes to a bit of study-specific Python code
+3. Record the structured results in an AWS Athena database
+
+Because Cumulus ETL has a growing internal library of NLP support code
+(things like automatically caching results, calling known interfaces like
+[Hugging Face's inference API](https://github.com/huggingface/text-generation-inference),
+or configuring cTAKES with a custom dictionary),
+the study-specific Python code can focus on the clinical logic.
+
+#### Example Code
+In pseudocode, here's the Python code for a task that talks to an LLM like Llama2 might look like:
+
+```python
+for clinical_note in etl.read_notes():
+    prompt = "Does the patient in this clinical note have a nosebleed?"
+    yield etl.ask_llama2(prompt, clinical_note)
+```
+
+Those calls to `etl.*` are calls to the internal NLP support code that the task does not have to
+re-invent.
+
+And with that relatively low level of complexity (though finding the right prompt can be hard),
+you've got a study task that you can run over all your institution's clinical notes. 
+
 ## Available NLP Strategies
 
+### Large Language Models (LLMs)
```

Comment:
Moved this section above cTAKES and talk more about local & cloud LLMs in general, to emphasize our support here.

Unfortunately, we don't actually have shipped code for a cloud LLM or this section could be stronger. But we can update it when we do.